services/controller/src/plz/controller/instances/docker.py

```diff
@@ -131,6 +138,8 @@ def release(self,
     def _do_release(self, results_storage: ResultsStorage):
         self.stop_execution()
         self._publish_results(results_storage)
+        # Check that we could collect the logs before destroying the container
+        results_storage.check_logs_available(self.execution_id)
```

Comment:
Check that the `finished_file` has been written, not the logs.

In the interface:

```python
    @abstractmethod
    def is_finished(self, execution_id: str):
        pass
```

In the `LocalResultsStorage` implementation:

```python
    def is_finished(self, execution_id: str):
        paths = Paths(self.directory, execution_id)
        return os.path.exists(paths.finished_file)
```